src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -671,9 +671,29 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
             $dumper->setProxyDumper(new ProxyDumper(md5($cache->getPath())));
         }
 
-        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'debug' => $this->debug));
+        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'as_files' => true, 'debug' => $this->debug));
```

Comment:
Because there are many benefits to splitting the container into several files.
ow does this relates to fopen any limit? Service files should be closed by PHP as soon as they are consumed, isn't it?

Reply:
@nicolas-grekas I meant the limit imposed by `/etc/security/limits.conf` (which includes all loaded files).

You maintain [here](https://github.com/symfony/symfony/pull/23678#issue-132477897) that you've "updated the Kernel to set that flag in non-debug envs", but this is not the case.

Is there any way to get around this option apart from overriding `Kernel:dumpContainer()` (which doesn't seem a good idea)?

Reply:
> /etc/security/limits.conf (which includes all loaded files)

I only see `nofile`, which limits *currently* open files. Doesn't PHP close them once executed, thus the culprit is elsewhere?

> get around this option

not really as this should work for everyone...

Reply:
It does close them, except if the file is autoloaded, and needs to autoload another file, in such a case, the chain will remain open until last statement

Reply:
We run the tests with coverage and thus with `phpdbg` which may be the curlpit, as you suggested...

Reply:
Yes, phpdbg has the max_open_file problem , I dont really know why , I guess it loads PHP files differently from other SAPI , I should have a look at the source code for that.

You should increase the limit by using ulimit -n

Reply:
@nicolas-grekas @jpaul 
Thank you for this feedback. 
Unfortunately `ulimits` are not available on Codeship.